Source/WebCore/Modules/mediastream/MediaStreamTrackProcessor.h

```diff
@@ -67,7 +67,8 @@ class MediaStreamTrackProcessor
         Source(Ref<MediaStreamTrack>&&, MediaStreamTrackProcessor&);
         ~Source();
 
-        bool isWaiting() const;
+        bool isWaiting() const { return m_isWaiting; }
+        bool isCancelled() const { return m_isCancelled; }
```

Comment:
some threading WTF_GUARDED_BY_CAPABILITY check would help reading stuff here.

Reply:
That would be nice indeed, not sure how that can be done, let me check

Reply:
`assertIsCurrent(instanceOfWorkerThread());`

Reply:
I'll add comments instead, `Source` is single threaded.
`VideoFrameObserverWrapper` is the one that lives in various threads.

Reply:
OK, I'll add WTF_GUARDED_BY_CAPABILITY in a follow-up as it might require adding some code to ScriptExecutionContext.

Reply:
I'll do that in https://github.com/WebKit/WebKit/pull/37488